include/seastar/core/sharded.hh

```diff
@@ -851,12 +852,14 @@ private:
     static future<> destroy_on(PtrType p, unsigned cpu) noexcept {
         if (p) {
             if (cpu != this_shard_id()) {
+              engine().run_in_background([cpu, p = std::move(p)] () mutable {
```

Comment:
This is not a background task, the future is (was) returned to the caller. 

Now this always returns a ready future, which is wrong.